### DISCUSSION TITLE
MIDI configuration randomly working/failing between reboots

### DISCUSSION THREAD

**Discussion by powerinpete** [GrandOrgue/grandorgue#1626] [Help] [answered]
Hi all, my problem is similar to [Thread#1616](https://github.com/GrandOrgue/grandorgue/discussions/1616) but will require a different solution.

In my case I am upgrading from an old (i3) computer running the last version of the old GO-LIVE distro. This has worked well for several years within the limitations of the old computer, OS and GO versions. The Allen organ console, in a church, has two MIDI controllers (one for keyboards, the other for stops/pistons) running through two identical MIDI-USB interfaces (Roland UM-ONE) into two usb ports. The new computer has a fresh install of the latest Ubuntu (Xubuntu 23.04) and the latest GO (3.12.2-1 [wx32]).

My problem with the new system is that my GO MIDI configuration will randomly work or not between boots. When the configuration fails I know the MIDI devices are still working because the MIDI send events will still operate the stop SAMs (I guess because my send events are sent to all interfaces regardless of ID) even while the keyboard and stop receive events do not work. I am assuming this is because either the USB names or MIDI IDs are changing between boots but I can't figure out why yet.

I am aware that Linux will randomly assign different USB names for two identical USB devices, like I have, when they are plugged in. But this should not be a problem if the same MIDI devices are permanently plugged into the same USB ports...and this has always been the case with my old system.

I am unsure if this is an Ubuntu or a GO issue. Thanks for any suggestions on how to diagnose it!

Peter

**Comment by larspalo** [comment]
Would it be possible to adjust all the expected MIDI signals to work without specific device information? If it works for the stops then it should be possible to do for manuals and pedals too, especially if the Allen console uses different MIDI channels for the manuals so that it's easy to differentiate between them.

In the old GO-LIVE before the updates done to the RtMidi when GO moved to GitHub we had a special patched version of RtMidi used in GO that fixed this kind of issues, but it's long gone and kind of replaced with the regex matching described in #1616.

**Reply by powerinpete** [reply]
It might be possible...if I knew how :-) . I know little about the MIDI side of things because my previous installations of GO have always "Just Worked". I was looking at the MIDI channel offset setting but didn't try it today as I need a working organ for church tomorrow and I left the computer running in a working state just to make sure.

If it helps here is what happens when GO MIDI boots up working:

- keyboards work as configured.
- when I move a stop on the console it lights up the stop on GO as configured in receive events.
- when I click a stop on GO it physically moves the stop on the console as configured in send events.
- pistons work as configured and physically move the console stops and light up the GO stops.

And when MIDI randomly doesn't work:

- keyboards and pistons don't work.
- moving a console stop will not move a GO stop
- clicking a GO stop will light it up and physically move the console stop as configured. But even then, clicking on a GO keyboard note will not produce a sound...which seems strange.

**Reply by powerinpete** [reply]
Can I configure receive events from any MIDI device the same as you do with send events (which is the default with no other choice)? As you point out I think stops and keys are indeed on different channels on the Allen.

Sorry...the church has limited internet access and is 25km away from my home so I can't quickly check or test these things.

**Reply by larspalo** [reply]
There seems to be quite a few things that need to be fixed in your setup. As the audio also stops working when the configuration fails it's not only MIDI related.

Generalizing the MIDI events might work in your favor. Right click on one of the elements and in the resulting dialog there will likely be some stuff filled in since last configuration. Test selecting "Any device" as device but keep everything else as it was before. Then OK to close the dialog. If it works for that element, try the others.

For audio devices the device enumeration can unfortunately also change with reboots. Sometimes it helps to select a default device, and it most certainly helps to go the extra mile and learn to use jack for audio. QjackCtrl is your friend for that.

**Reply by powerinpete** [reply]
Thanks for that. I'll do some more testing today.

**Reply by powerinpete** [reply, marked as answer]
After testing today the problem was indeed that the OS was randomly swapping the names of the two MIDI devices. Changing the midi device settings of all the keyboard, stop and piston receive events from a specific midi device name to "Any device" solved the problem. Thanks for the suggestion!

The audio problem is coincidental. Xubuntu occasionally fails to pick up the audio device on boot and instead assigns the sound to "Dummy output". Oh well...more testing. :-P
